packages/backend-data/src/convert_authorization_modes.ts

```diff
@@ -146,6 +164,7 @@ const computeApiKeyAuthFromResource = (
   providedAuthConfig: ProvidedAuthConfig | undefined,
   authModes: AuthorizationModes | undefined
 ): CDKApiKeyAuthorizationConfig | undefined => {
+  // See also computeDefaultAuthorizationMode.
   if (providedAuthConfig || authModes) {
     return;
   }
```

Comment:
This should move to `isUsingDefaultApiKeyAuth` (and perhaps renamed that function to `useDefaultApiKeyAuth`) and use it here and in `computeDefaultAuthorizationMode`.

Reply:
Moving to `isUsingDefaultApiKeyAuth` might be hard due to https://github.com/aws-amplify/amplify-backend/blob/b0112e34f94b05179addf67c213a5cfce16dc079/packages/backend-data/src/factory.ts#L186-L189 .

However I like the idea to extract common piece. I'll see what can be done.